2025-04-23 08:19:15 - **** Job starts ****
2025-04-23 08:19:15 - **** Quest info ****
User: owb0346
Job id: 731592
Job name: simu_genotypes
Node name: 
Hostname: qnode0397
Task id: 2
2025-04-23 08:19:15 - **** Loading modules ****

Currently Loaded Modules:
  1) bzip2/1.0.8-gcc-4.8.5   2) htslib/1.16

 

2025-04-23 08:19:31 - **** Loading mamba environment ****
2025-04-23 08:19:44 - **** Run simulation ****
[    INFO] Using seed 20240126 (sim_genotype.py:359)
[    INFO] Simulating generation 1 (sim_genotype.py:457)
[    INFO] Simulating generation 2 (sim_genotype.py:463)
[    INFO] Simulating generation 3 (sim_genotype.py:463)
[    INFO] Simulating generation 4 (sim_genotype.py:463)
[    INFO] Simulating generation 5 (sim_genotype.py:463)
[    INFO] Simulating generation 6 (sim_genotype.py:463)
[    INFO] Simulating generation 7 (sim_genotype.py:463)
[    INFO] Simulating generation 8 (sim_genotype.py:463)
[    INFO] Simulating generation 9 (sim_genotype.py:463)
[    INFO] Simulating generation 10 (sim_genotype.py:463)
[    INFO] Outputting breakpoint file ./chr2.bp (sim_genotype.py:503)
[    INFO] Outputting file ./chr2.vcf.gz (sim_genotype.py:75)
[ WARNING] The max_variants parameter was not specified. We have no choice but to append to an ever-growing array, which can lead to memory overuse! (genotypes.py:165)
[    INFO] Loading genotypes from 3202 samples (genotypes.py:289)
[    INFO] Copying 6088598 variants into np arrays. (genotypes.py:174)
[    INFO] Transposing genotype matrix of size (6088598, 3202, 3) (genotypes.py:209)
[    INFO] Writing VCF records (genotypes.py:779)
Traceback (most recent call last):
  File "/projects/p32505/opt/env/AI_env/bin/haptools", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/gpfs/projects/p32505/opt/env/AI_env/lib/python3.12/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/projects/p32505/opt/env/AI_env/lib/python3.12/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/gpfs/projects/p32505/opt/env/AI_env/lib/python3.12/site-packages/click/core.py", line 1697, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/projects/p32505/opt/env/AI_env/lib/python3.12/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/projects/p32505/opt/env/AI_env/lib/python3.12/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/projects/p32505/opt/env/AI_env/lib/python3.12/site-packages/haptools/__main__.py", line 298, in simgenotype
    output_vcf(
  File "/gpfs/projects/p32505/opt/env/AI_env/lib/python3.12/site-packages/haptools/sim_genotype.py", line 223, in output_vcf
    gts.write()
  File "/gpfs/projects/p32505/opt/env/AI_env/lib/python3.12/site-packages/haptools/data/genotypes.py", line 797, in write
    record.samples[sample]["GT"] = tuple(
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "pysam/libcbcf.pyx", line 3540, in pysam.libcbcf.VariantRecordSample.__setitem__
  File "pysam/libcbcf.pyx", line 872, in pysam.libcbcf.bcf_format_set_value
  File "pysam/libcbcf.pyx", line 671, in pysam.libcbcf.bcf_encode_alleles
ValueError: Invalid allele index
2025-04-23 11:33:52 - **** Job ends ****
